HealthVault/Classes/Clients/MHVThingClient.m

```diff
@@ -447,6 +455,25 @@ - (void)createNewThings:(MHVThingCollection *)things
          MHVThingKeyCollection *keys = [self thingKeyResultsFromResponse:response];
          
 #ifdef THING_CACHE
+         
+         // If the connection is offline cache the pending request.
+         if ([self isOfflineError:error])
+         {
+             [self cachePendingMethod:method
+                               things:things
+                        operationType:MHVThingOpertaionTypeCreate
+                         networkError:error
+                          completion:^(NSError * _Nullable error)
+             {
+                 if (completion)
+                 {
+                     completion(keys, error);
```

Comment:
nit, keys looks like it will be nil because there wasn't a response, but maybe pass nil to be clearer